src/Microsoft.TemplateEngine.Cli/ExtendedTemplateEngineHost.cs

```diff
@@ -144,5 +146,26 @@ private bool GlobalJsonFileExistsInPath
                 return found;
             }
         }
+
+        public void RebuildCacheFromSettingsIfNotCurrent(IEngineEnvironmentSettings environmentSettings)
+        {
+            TemplateCache cache = new TemplateCache(environmentSettings);
+            cache.LoadTemplateCacheForLocale(Locale, out string cacheVersion);
+
+            if (TemplateCache.CheckIfCacheVersionIsCurrent(cacheVersion))
```

Comment:
This feels like something that the cache could do internally (`cache.IsCurrentVersion` or some such thing)